plat/nvidia/tegra/soc/t186/drivers/smmu/smmu.c

```diff
@@ -438,18 +444,24 @@ void tegra_smmu_save_context(void)
 	assert(!((pgshift != PGSHIFT) || (cb_size != CB_SIZE)));
 #endif
 
+	assert((smmu_ctx_addr >= tzdram_base) && (smmu_ctx_addr <= tzdram_end));
```

Comment:
@davidcunado-arm 

We just want to make sure that the address being used to store the context is a valid TZDRAM address. But I see what you are saying. I will try to pass the destination size to the smmu function as well. Need some time to figure out, so will push another PR to fix this.